mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOpsTypes.h

```diff
@@ -502,6 +502,16 @@ class GroupType
   llvm::ArrayRef<mlir::Type> getBody() const;
 };
 
+/// Return true if the given \p Ty is a SYCL type.
+inline bool isSYCLType(Type Ty) {
```

Comment:
Seems like a useful refactoring. 

Would it be possible to have avoid explicitly listing all types here in a future PR by using a common MLIR type trait for all SYCL types?

Reply:
There are currently no common MLIR type traits for all SYCL types, that's unique to SYCL types. Are you suggesting adding an empty one and use it for this purpose?

Reply:
Resolved by https://github.com/intel/llvm/pull/7171.